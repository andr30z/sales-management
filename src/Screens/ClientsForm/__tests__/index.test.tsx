import { fireEvent, render, waitFor } from "@testing-library/react-native";
import React from "react";
import { act } from "react-test-renderer";
import {
  EvaApplicationProvider,
  initialMockClient,
  MockedSalesInfoContext,
  MockedStackNavigator,
  MockedToastProvider,
} from "../../../TestCommons";
import {
  ClientsForm,
  CLIENT_FORM_CONTAINER_TEST_ID,
  CLIENT_FORM_PHONE_INPUT,
  CLIENT_FORM_SUBMIT_BTN,
} from "../index";

const mockedNavigate = jest.fn();
const mockedGoBack = jest.fn();
jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,

    useNavigation: () => ({
      navigate: mockedNavigate,
      dispatch: jest.fn(),
      goBack: mockedGoBack,
    }),
  };
});
describe("<ClientsForm />", () => {
  function TestClientForm(props: { id?: string }) {
    return (
      <MockedSalesInfoContext>
        <MockedToastProvider>
          <EvaApplicationProvider>
            <MockedStackNavigator
              params={props}
              component={ClientsForm as any}
            />
          </EvaApplicationProvider>
        </MockedToastProvider>
      </MockedSalesInfoContext>
    );
  }

  it("should render correctly", () => {
    const { toJSON, getByTestId } = render(<TestClientForm />);

    expect(toJSON()).toMatchSnapshot();
    expect(getByTestId(CLIENT_FORM_CONTAINER_TEST_ID)).toBeDefined();
  });

  it("should go back to previous screen when submitted", async () => {
    const { getByTestId } = render(
      <TestClientForm id={initialMockClient.id} />
    );

    const phoneInput = getByTestId(CLIENT_FORM_PHONE_INPUT);
    fireEvent.changeText(phoneInput, "5561992999156806");

    const submitBtn = getByTestId(CLIENT_FORM_SUBMIT_BTN);
    fireEvent.press(submitBtn);
    await waitFor(() => {
      expect(mockedGoBack).toBeCalledTimes(1);
    });
  });
});
