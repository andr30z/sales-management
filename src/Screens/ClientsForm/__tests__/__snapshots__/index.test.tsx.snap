// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClientsForm /> should render correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        [
          {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "rgb(242, 242, 242)",
              "flex": 1,
            },
            undefined,
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="box-none"
          style={
            {
              "zIndex": 1,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            onLayout={[Function]}
            pointerEvents="box-none"
            style={null}
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "zIndex": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "rgb(255, 255, 255)",
                    "borderBottomColor": "rgb(216, 216, 216)",
                    "flex": 1,
                    "shadowColor": "rgb(216, 216, 216)",
                    "shadowOffset": {
                      "height": 0.5,
                      "width": 0,
                    },
                    "shadowOpacity": 0.85,
                    "shadowRadius": 0,
                  }
                }
              />
            </View>
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "height": 44,
                  "maxHeight": undefined,
                  "minHeight": undefined,
                  "opacity": undefined,
                  "transform": undefined,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  {
                    "height": 0,
                  }
                }
              />
              <View
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "stretch",
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  collapsable={false}
                  pointerEvents="box-none"
                  style={
                    {
                      "alignItems": "flex-start",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "justifyContent": "center",
                      "marginStart": 0,
                      "opacity": 1,
                    }
                  }
                />
                <View
                  collapsable={false}
                  pointerEvents="box-none"
                  style={
                    {
                      "justifyContent": "center",
                      "marginHorizontal": 16,
                      "maxWidth": 718,
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    accessibilityRole="header"
                    aria-level="1"
                    collapsable={false}
                    numberOfLines={1}
                    onLayout={[Function]}
                    style={
                      {
                        "color": "rgb(28, 28, 30)",
                        "fontSize": 17,
                        "fontWeight": "600",
                      }
                    }
                  >
                    MockedScreen
                  </Text>
                </View>
                <View
                  collapsable={false}
                  pointerEvents="box-none"
                  style={
                    {
                      "alignItems": "flex-end",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "justifyContent": "center",
                      "marginEnd": 0,
                      "opacity": 1,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <RNSScreenContainer
          onLayout={[Function]}
          style={
            {
              "flex": 1,
            }
          }
        >
          <RNSScreen
            activityState={2}
            collapsable={false}
            forwardedRef={[Function]}
            gestureResponseDistance={
              {
                "bottom": -1,
                "end": -1,
                "start": -1,
                "top": -1,
              }
            }
            pointerEvents="box-none"
            style={
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "opacity": 1,
                }
              }
            />
            <View
              accessibilityElementsHidden={false}
              closing={false}
              gestureVelocityImpact={0.3}
              importantForAccessibility="auto"
              onClose={[Function]}
              onGestureBegin={[Function]}
              onGestureCanceled={[Function]}
              onGestureEnd={[Function]}
              onOpen={[Function]}
              onTransition={[Function]}
              pointerEvents="box-none"
              style={
                [
                  {
                    "display": "flex",
                    "overflow": undefined,
                  },
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                ]
              }
              transitionSpec={
                {
                  "close": {
                    "animation": "spring",
                    "config": {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 10,
                      "restSpeedThreshold": 10,
                      "stiffness": 1000,
                    },
                  },
                  "open": {
                    "animation": "spring",
                    "config": {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 10,
                      "restSpeedThreshold": 10,
                      "stiffness": 1000,
                    },
                  },
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "flex": 1,
                  }
                }
              >
                <View
                  collapsable={false}
                  forwardedRef={[Function]}
                  handlerTag={1}
                  handlerType="PanGestureHandler"
                  needsOffscreenAlphaCompositing={false}
                  onGestureHandlerEvent={[Function]}
                  onGestureHandlerStateChange={[Function]}
                  style={
                    {
                      "flex": 1,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                        {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    style={
                      [
                        {
                          "flex": 1,
                          "overflow": "hidden",
                        },
                        [
                          {
                            "backgroundColor": "rgb(242, 242, 242)",
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    <View
                      style={
                        {
                          "flex": 1,
                          "flexDirection": "column-reverse",
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          onLayout={[Function]}
                          style={
                            [
                              {
                                "flex": 1,
                              },
                              {
                                "paddingBottom": 0,
                              },
                            ]
                          }
                        >
                          <RCTScrollView
                            contentContainerStyle={
                              {
                                "flex": 1,
                                "paddingBottom": 40,
                              }
                            }
                            keyboardShouldPersistTaps="handled"
                            style={
                              {
                                "backgroundColor": "white",
                                "flex": 1,
                              }
                            }
                          >
                            <View>
                              <View
                                style={
                                  {
                                    "backgroundColor": "#fff",
                                    "flex": 1,
                                    "flexDirection": "column",
                                    "justifyContent": "center",
                                    "paddingHorizontal": 15,
                                    "width": "100%",
                                  }
                                }
                                testID="CONTAINER_TEST_ID"
                              >
                                <View
                                  style={
                                    {
                                      "alignItems": "center",
                                      "flex": null,
                                      "justifyContent": "center",
                                      "minHeight": 100,
                                    }
                                  }
                                >
                                  <Text
                                    appearance="default"
                                    category="h2"
                                    status="primary"
                                    style={
                                      [
                                        {
                                          "color": "#FF1173",
                                          "fontFamily": "System",
                                          "fontSize": 32,
                                          "fontWeight": "800",
                                        },
                                        [
                                          {
                                            "textAlign": "center",
                                          },
                                          {
                                            "fontFamily": "Satisfy_400Regular",
                                          },
                                        ],
                                      ]
                                    }
                                  >
                                    Cadastro de Cliente 
                                  </Text>
                                </View>
                                <View
                                  accessible={true}
                                  collapsable={false}
                                  focusable={false}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    {
                                      "marginVertical": 15,
                                      "opacity": 1,
                                      "width": "100%",
                                    }
                                  }
                                >
                                  <Text
                                    appearance="default"
                                    category="p1"
                                    style={
                                      [
                                        {
                                          "color": "#222B45",
                                          "fontFamily": "System",
                                          "fontSize": 15,
                                          "fontWeight": "400",
                                        },
                                        [
                                          {
                                            "color": "#8F9BB3",
                                            "fontFamily": "System",
                                            "fontSize": 12,
                                            "fontWeight": "800",
                                            "marginBottom": 4,
                                          },
                                          {
                                            "textAlign": "left",
                                          },
                                        ],
                                      ]
                                    }
                                  >
                                    Nome do cliente
                                  </Text>
                                  <View
                                    style={
                                      [
                                        {
                                          "backgroundColor": "#F7F9FC",
                                          "borderColor": "#E4E9F2",
                                          "borderRadius": 4,
                                          "borderWidth": 1,
                                          "minHeight": 40,
                                          "paddingHorizontal": 8,
                                          "paddingVertical": 7,
                                        },
                                        {
                                          "alignItems": "center",
                                          "flexDirection": "row",
                                          "width": "100%",
                                        },
                                      ]
                                    }
                                  >
                                    <TextInput
                                      appearance="default"
                                      editable={true}
                                      onBlur={[Function]}
                                      onChangeText={[Function]}
                                      onFocus={[Function]}
                                      onMouseEnter={[Function]}
                                      onMouseLeave={[Function]}
                                      placeholder="Ex: Zezinho 123"
                                      placeholderTextColor="#8F9BB3"
                                      size="medium"
                                      status="basic"
                                      style={
                                        [
                                          {
                                            "color": "#222B45",
                                            "fontFamily": "System",
                                            "fontSize": 15,
                                            "fontWeight": "normal",
                                            "marginHorizontal": 8,
                                          },
                                          {
                                            "flexBasis": "auto",
                                            "flexGrow": 1,
                                            "flexShrink": 1,
                                          },
                                          null,
                                          undefined,
                                        ]
                                      }
                                      value=""
                                    />
                                  </View>
                                </View>
                                <View
                                  accessible={true}
                                  collapsable={false}
                                  focusable={false}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    {
                                      "marginVertical": 15,
                                      "opacity": 1,
                                      "width": "100%",
                                    }
                                  }
                                >
                                  <Text
                                    appearance="default"
                                    category="p1"
                                    style={
                                      [
                                        {
                                          "color": "#222B45",
                                          "fontFamily": "System",
                                          "fontSize": 15,
                                          "fontWeight": "400",
                                        },
                                        [
                                          {
                                            "color": "#8F9BB3",
                                            "fontFamily": "System",
                                            "fontSize": 12,
                                            "fontWeight": "800",
                                            "marginBottom": 4,
                                          },
                                          {
                                            "textAlign": "left",
                                          },
                                        ],
                                      ]
                                    }
                                  >
                                    Telefone
                                  </Text>
                                  <View
                                    style={
                                      [
                                        {
                                          "backgroundColor": "#F7F9FC",
                                          "borderColor": "#E4E9F2",
                                          "borderRadius": 4,
                                          "borderWidth": 1,
                                          "minHeight": 40,
                                          "paddingHorizontal": 8,
                                          "paddingVertical": 7,
                                        },
                                        {
                                          "alignItems": "center",
                                          "flexDirection": "row",
                                          "width": "100%",
                                        },
                                      ]
                                    }
                                  >
                                    <TextInput
                                      appearance="default"
                                      editable={true}
                                      keyboardType="number-pad"
                                      maxLength={14}
                                      onBlur={[Function]}
                                      onChangeText={[Function]}
                                      onFocus={[Function]}
                                      onMouseEnter={[Function]}
                                      onMouseLeave={[Function]}
                                      placeholder="5561999999999"
                                      placeholderTextColor="#8F9BB3"
                                      size="medium"
                                      status="basic"
                                      style={
                                        [
                                          {
                                            "color": "#222B45",
                                            "fontFamily": "System",
                                            "fontSize": 15,
                                            "fontWeight": "normal",
                                            "marginHorizontal": 8,
                                          },
                                          {
                                            "flexBasis": "auto",
                                            "flexGrow": 1,
                                            "flexShrink": 1,
                                          },
                                          null,
                                          undefined,
                                        ]
                                      }
                                      value=""
                                    />
                                  </View>
                                </View>
                                <View
                                  accessible={true}
                                  collapsable={false}
                                  focusable={false}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    {
                                      "marginVertical": 15,
                                      "opacity": 1,
                                      "width": "100%",
                                    }
                                  }
                                >
                                  <Text
                                    appearance="default"
                                    category="p1"
                                    style={
                                      [
                                        {
                                          "color": "#222B45",
                                          "fontFamily": "System",
                                          "fontSize": 15,
                                          "fontWeight": "400",
                                        },
                                        [
                                          {
                                            "color": "#8F9BB3",
                                            "fontFamily": "System",
                                            "fontSize": 12,
                                            "fontWeight": "800",
                                            "marginBottom": 4,
                                          },
                                          {
                                            "textAlign": "left",
                                          },
                                        ],
                                      ]
                                    }
                                  >
                                    Observação
                                  </Text>
                                  <View
                                    style={
                                      [
                                        {
                                          "backgroundColor": "#F7F9FC",
                                          "borderColor": "#E4E9F2",
                                          "borderRadius": 4,
                                          "borderWidth": 1,
                                          "minHeight": 40,
                                          "paddingHorizontal": 8,
                                          "paddingVertical": 7,
                                        },
                                        {
                                          "alignItems": "center",
                                          "flexDirection": "row",
                                          "width": "100%",
                                        },
                                      ]
                                    }
                                  >
                                    <TextInput
                                      appearance="default"
                                      editable={true}
                                      multiline={true}
                                      numberOfLines={4}
                                      onBlur={[Function]}
                                      onChangeText={[Function]}
                                      onFocus={[Function]}
                                      onMouseEnter={[Function]}
                                      onMouseLeave={[Function]}
                                      placeholder="Observação sobre o cliente. Ex: 'Fulano é Caloteiro.'"
                                      placeholderTextColor="#8F9BB3"
                                      size="medium"
                                      status="basic"
                                      style={
                                        [
                                          {
                                            "color": "#222B45",
                                            "fontFamily": "System",
                                            "fontSize": 15,
                                            "fontWeight": "normal",
                                            "marginHorizontal": 8,
                                          },
                                          {
                                            "flexBasis": "auto",
                                            "flexGrow": 1,
                                            "flexShrink": 1,
                                          },
                                          null,
                                          undefined,
                                        ]
                                      }
                                      value=""
                                    />
                                  </View>
                                </View>
                                <View
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    {
                                      "alignItems": "center",
                                      "backgroundColor": "#FF1173",
                                      "borderColor": "#FF1173",
                                      "borderRadius": 4,
                                      "borderWidth": 1,
                                      "flexDirection": "row",
                                      "justifyContent": "center",
                                      "minHeight": 40,
                                      "minWidth": 40,
                                      "opacity": 1,
                                      "paddingHorizontal": 10,
                                      "paddingVertical": 12,
                                    }
                                  }
                                  testID="SUBMIT_BTN_FORM"
                                >
                                  <Text
                                    appearance="default"
                                    category="p1"
                                    style={
                                      [
                                        {
                                          "color": "#222B45",
                                          "fontFamily": "System",
                                          "fontSize": 15,
                                          "fontWeight": "400",
                                        },
                                        {
                                          "color": "#FFFFFF",
                                          "fontFamily": "System",
                                          "fontSize": 14,
                                          "fontWeight": "bold",
                                          "marginHorizontal": 10,
                                        },
                                      ]
                                    }
                                  >
                                    Cadastrar
                                  </Text>
                                </View>
                              </View>
                            </View>
                          </RCTScrollView>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RNSScreen>
        </RNSScreenContainer>
      </View>
    </RNCSafeAreaProvider>
  </View>
</View>
`;
